td/tracking/views.py

```diff
@@ -273,6 +323,144 @@ def set_event_number(self):
         Event.objects.filter(pk=self.object.id).update(number=(latest + 1))
 
 
+class EventUpdateView(LoginRequiredMixin, UpdateView):
+    model = Event
+    form_class = EventForm
+    template_name_suffix = "_update_form"
+
+    # Overwritten to include custom data
+    def get_context_data(self, *args, **kwargs):
+        context = super(EventUpdateView, self).get_context_data(**kwargs)
+        context["translators"] = self.get_translator_data(self)
+        context["facilitators"] = self.get_facilitator_data(self)
+        context["materials"] = self.get_material_data(self)
+        return context
+
+    # Overwritten to execute custom save and redirect upon valid submission
+    def form_valid(self, form):
+        event = self.object = form.save()
+
+        # Update translators info
+        translators = self.get_translator_data(self)
+        translator_ids = self.get_translator_ids(translators)
+        event.translators.clear()
+        event.translators.add(*list(Translator.objects.filter(id__in=translator_ids)))
+
+        # Add facilitators info
+        facilitators = self.get_facilitator_data(self)
+        facilitator_ids = self.get_facilitator_ids(facilitators)
+        event.facilitators.clear()
+        event.facilitators.add(*list(Facilitator.objects.filter(id__in=facilitator_ids)))
+
+        # Add materials info
+        materials = self.get_material_data(self)
+        material_ids = self.get_material_ids(materials)
+        event.materials.clear()
+        event.materials.add(*list(Material.objects.filter(id__in=material_ids)))
+
+        return redirect("tracking:charter_add_success", obj_type="event", pk=self.object.id)
+
+    # ----------------------------------- #
+    #    EVENTADDVIEW CUSTOM FUNCTIONS    #
+    # ----------------------------------- #
+
+    # Function: Returns an array of Translator objects' properties
+    def get_translator_data(self, form):
```

Comment:
I think it would have been handled by form processing if I use the `inlineformset` thing. At the moment, I'm still groping in the dark with `formsets` and `inlineformsets`, so I put a jQuery work around. And since those input is technically outside of Django, I have to process it manually. I'm really open to suggestions on how to improve this. Right now, I'm planning to refactor this into `inlineformsets` once I know how to do it.